app/services/concerns/xccdf/rule_group_rules.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module Xccdf
+  # Methods related to saving RuleGroupRules
+  module RuleGroupRules
+    extend ActiveSupport::Concern
+
+    included do
+      def save_rule_group_rules
+        @required_rule_group_rules = rule_group_rules_with_requires +
+                                     rule_group_rules_with_conflicts
+        @required_rule_group_rules = @required_rule_group_rules.compact
+
+        ::RuleGroupRule.transaction do
+          ::RuleGroupRule.import!(
+            @required_rule_group_rules.select(&:new_record?), on_duplicate_key_ignore: true
+          )
+
+          # rule_base = ::RuleGroupRule.joins(rule: :benchmark)
+          # rule_group_base = ::RuleGroupRule.joins(rule_group: :benchmark)
+
+          # links_to_remove(rule_base).delete_all
+          # links_to_remove(rule_group_base).delete_all
+
+        end
+      end
+
+      private
+
+      def rule_group_rules_with_requires
+        @rule_groups_with_requires ||= @op_rule_groups.flat_map do |op_rule_group|
+          selectable = rule_or_rule_group_for(ref_id: op_rule_group.id)
+          op_rule_group.requires.map do |required_ref_id|
+            requirable = rule_or_rule_group_for(ref_id: required_ref_id)
+            ::RuleGroupRule.find_or_initialize_by(
+              selectable_id: selectable[0]&.id, selectable_type: selectable[1],
+              requirable_id: requirable[0]&.id, requirable_type: requirable[1]
+            )
+          end
+        end
+      end
+
+      def rule_group_rules_with_conflicts
+        @rule_groups_with_conflicts ||= @op_rule_groups.flat_map do |op_rule_group|
+          selectable = rule_or_rule_group_for(ref_id: op_rule_group.id)
+          conflictable = rule_or_rule_group_for(ref_id: op_rule_group.conflicts)
+          ::RuleGroupRule.find_or_initialize_by(
+            selectable_id: selectable[0]&.id, selectable_type: selectable[1],
+            conflictable_id: conflictable[0]&.id, conflictable_type: conflictable[1]
+          )
+        end
+      end
```

Comment:
@skateman using polymorphic associations helped with the complexity a lot but I still split conflicts and requires into 2 different methods here so that I could do `::RuleGroupRule.find_or_initialize_by` on the required rule/rule_groups and conflicting rule/rule_groups. This seems to be working....

Reply:
I missed this comment yesterday but I'll try this today. I think this could would too